### PR TITLE
[pulsar-operator] Support deploying operators with namespace-scoped permissions

### LINKED ISSUE #661: Support deploying operators with namespace-scoped permissions

### PR DESCRIPTION
**Pull request by FushuWang**
Resolve #661 
### Motivation

Add support for watching a specific namespace with namespace-scoped permissions.

### Modifications

- Add a new value item`watchNamespace` to specify a namespace to watch.
- Add parameter `--watch-namespace` for the operator container args in the deployment.
- Add a `Role`  in the specified namespace.
- Add a `RoleBinding` for the operator service account in the specified namespace.
- Remove duplicated role content, use `serviceAccount.clusterRole` to control `Kind` in one rbac manifest

### Verifying this change
if `watchNaespace` is empty and `clusterRole` is `false`, the operators will watch the namespace that operators are deployed into.
- Set watchNamespace.
- Disable `clusterRole`  for the operators.
```yaml
watchNamespace: test
pulsar:
  serviceAccount:
    clusterRole: false
bookkeeper:
  serviceAccount:
    clusterRole: false
zookeeper:
  serviceAccount:
    clusterRole: false
```

- [x] Make sure that the change passes the CI checks.

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required`

### CODE REVIEW

File: charts/pulsar-operator/templates/bookkeeper-operator/operator_rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+metadata:
+  name: '{{ template "pulsar.fullname" . }}-{{ .Values.bookkeeper.rbac.name }}-operator'
+  {{- if not .Values.bookkeeper.serviceAccount.clusterRole }}
+  namespace: {{ template "pulsar.namespace" .}}
+  {{- end }}
+  labels:
+    {{- include "pulsar.standardLabels" . | nindent 4 }}
+    components: {{ .Values.bookkeeper.component }}
+rules:
+- apiGroups:
```
Comment (MarvinCai) [inline charts/pulsar-operator/templates/bookkeeper-operator/operator_rbac.yaml:32]: is the permission enough for operator to run?

Comment (FushuWang) [reply charts/pulsar-operator/templates/bookkeeper-operator/operator_rbac.yaml:32]: > is the permission enough for operator to run?

It looks good now, I tried to install a release of sn-platform, but in the meantime didn't find another permission error in operator's log.

File: charts/pulsar-operator/templates/pulsar-operator/operator.yaml [hunk 2]
```diff
@@ -49,6 +49,7 @@ spec:
         args:
         - --metrics-addr=127.0.0.1:8080
         - --enable-leader-election
+        - --watch-namespace={{ .Values.watchNamespace }}
```
Comment (fantapsody) [inline charts/pulsar-operator/templates/pulsar-operator/operator.yaml:52]: If the watch namespace is not set, but the clusterRole is false, the operator should watch its own namespace?

Comment (FushuWang) [reply charts/pulsar-operator/templates/pulsar-operator/operator.yaml:52]: > If the watch namespace is not set, but the clusterRole is false, the operator should watch its own namespace?

yeah, my bad. I have did it at the role namespace, forgot to update here.

Comment (FushuWang) [reply charts/pulsar-operator/templates/pulsar-operator/operator.yaml]: Done.

**Review by MarvinCai — APPROVED** [review]

**Pull request merged**